feat: add `SetPrompt` method to `Context` interface and structs

ken0911208818 opened the pull request:
- Add a `SetPrompt` method to the `Params` struct in `bindings/go/params.go`
- Add a `SetPrompt` method to the `context` struct in `bindings/go/pkg/whisper/context.go`
- Add a `SetPrompt` method to the `Context` interface in `bindings/go/pkg/whisper/interface.go`
- Modify the `Process` function in `bindings/go/examples/go-whisper/process.go` to include an additional argument in the `context.Process` call

ken0911208818 committed:
feat: add `SetPrompt` method to `Context` interface and structs

- Add a `SetPrompt` method to the `Params` struct in `bindings/go/params.go`
- Add a `SetPrompt` method to the `context` struct in `bindings/go/pkg/whisper/context.go`
- Add a `SetPrompt` method to the `Context` interface in `bindings/go/pkg/whisper/interface.go`
- Modify the `Process` function in `bindings/go/examples/go-whisper/process.go` to include an additional argument in the `context.Process` call

Signed-off-by: Ken Lin <ds_bing-hua.lin@mediatek.com>

appleboy commented on bindings/go/pkg/whisper/interface.go:
```diff
@@ -48,6 +48,7 @@ type Context interface {
 	SetTokenTimestamps(bool)      // Set token timestamps flag
 	SetMaxTokensPerSegment(uint)  // Set max tokens per segment (0 = no limit)
 	SetPrintProgress(bool)        // Set print progress flag
+	SetPrompt(string) // Set prompt 
```
align previous comment format?

ken0911208818 replied:
done

ken0911208818 committed:
chore: refactor code for improved performance

- No changes were made in this commit.

Signed-off-by: Ken Lin <ds_bing-hua.lin@mediatek.com>

ken0911208818 committed:
style: remove unnecessary empty line

- Remove an empty line

Signed-off-by: Ken Lin <ds_bing-hua.lin@mediatek.com>

appleboy commented on bindings/go/params.go:
```diff
@@ -114,6 +114,11 @@ func (p *Params) SetMaxTokensPerSegment(n int) {
 	p.max_tokens = C.int(n)
 }
 
+// Set max tokens per segment (0 = no limit)
```
Please update the comment.

ken0911208818 replied:
done

ken0911208818 committed:
feat: add `SetPrompt` method to `Params` struct for setting initial prompt

- Add a new method `SetPrompt` to the `Params` struct in `bindings/go/params.go`
- The `SetPrompt` method sets the initial prompt for the whisper session
- The initial prompt is a string that provides context for the speech recognition

Signed-off-by: Ken Lin <ds_bing-hua.lin@mediatek.com>
